File: OOP/drama.cpp
#include "drama.h"
#include <cstdlib>

char *ReadString(FILE *file);
char *RandomString();

drama::~drama() {
    free(director);
}

// Read drama from file.
void drama::Read(FILE *file) {
    film::Read(file);
    director = ReadString(file);
}

// Write drama to file.
void drama::Write(FILE *file) {
    fprintf(file, "%s", "drama ");
    film::Write(file);
    fprintf(file, "%s%s", director, " ");
}

void drama::Random() {
	film::Random();
	director = RandomString();
}

File: Dynamic/Drama.py
from Film import Film
from Utilities import *


class Drama(Film):
    def __init__(self):
        super().__init__()
        self.director = ''

    def read(self, params):
        super().read(params)
        self.director = params[3]

    def to_string(self):
        return 'drama ' + super().to_string() + ' ' + self.director

    def random(self):
        super().random()
        self.director = random_string()
